lib/psi.rb

```diff
@@ -416,28 +722,612 @@ def processTBDinputs(surfaces, edges, set, io_path = nil, schema_path = nil)
             n += 1
             if edge.has_key?(:psi)                                    # optional
               p = edge[:psi]
-              raise "PSI mismatch" unless psi.set.has_key?(p)
-              raise "#{p} missing PSI #{t}" unless psi.set[p].has_key?(t)
+              raise "PSI mismatch (TBD inputs)" unless psi.set.has_key?(p)
+              #holds, values = psi.shorthands(p)
+              #next if holds.empty?
+              #next if values.empty?
+              tt = psi.safeType(p, t)
+              #tt = safeType(t, holds)
+              raise "#{p} missing PSI #{t} or variants" if tt.nil?
               e[:io_set] = p
             end
           end
         end
-        if n == 0
-          puts edge[:surfaces]
-          raise "Edge: missing OpenStudio match"
-        end
+        raise "Edge: missing OpenStudio match" if n == 0
       end
     end
   else
     # No (optional) user-defined TBD JSON input file.
     # In such cases, "set" must refer to a valid PSI set
     raise "Incomplete PSI set #{set}" unless psi.complete?(set)
-    io[:unit] = [{ psi: set }]                 # i.e. default PSI set & no KHI's
+    io[:building] = [{ psi: set }]             # i.e. default PSI set & no KHI's
   end
-
   return io, psi, khi
 end
 
+##
+# Return min & max values for schedule (ruleset).
+#
+# @param [OpenStudio::Model::ScheduleRuleset] sched An OS schedule (ruleset)
+#
+# @return [Hash] :min & :max; nilled if invalid.
+def scheduleRulesetMinMax(sched)
```

Comment:
This, and several other related functions, allow TBD to distinguish unenclosed/unconditioned vs semi-heated/conditioned spaces/thermal zones, attics, plenums, etc., IF required inputs are present in an OSM.